File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    // create a stack
    // create an object look up
    // loop through s
    // if char is in brackets
    // add corresponding value to stack
    // if next char matches top of stack, pop

    let stack = [];
    const brackets = { "(":")", "[":"]", "{":"}" };

    for (char of s) {
        if (char in brackets) {
            stack.push(brackets[char])
        } else {
            if (stack.length > 0 && char === stack[stack.length-1]) {
                stack.pop();
            } else {
                return false
            }
        }
    }
    return stack.length === 0;
    
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []

        dict = {
            "(":")",
            "{":'}',
            "[": ']'

        }


        for char in s:
            if char in dict:
                stack.append(char)
            elif len(stack) == 0 or char != dict[stack.pop()]:
                return False


      


        
        return len(stack) == 0
